python/ray/ml/checkpoint.py

```diff
@@ -492,3 +538,22 @@ def _unpack(stream: bytes, path: str) -> str:
     with tarfile.open(fileobj=io.BytesIO(stream)) as tar:
         tar.extractall(path)
     return path
+
+
+def _get_del_lock_path(path: str) -> str:
+    """Get the path to the deletion lock file."""
+    return os.path.join(path, f".del_lock_{os.getpid()}")
+
+
+def _make_dir(path: str, drop_del_lock: bool = True) -> None:
+    """Create the temporary checkpoint dir in ``path``."""
+    os.makedirs(path, exist_ok=True)
+    # Drop marker
```

Comment:
If we put it outside the dir, then you can create it without holding the file lock / before the dir exists.